### ISSUE TITLE
Scheduler for preference comparison gathering

### ISSUE THREAD

**Issue by ejnnr** [HumanCompatibleAI/imitation#390]
We currently gather one big chunk of preference comparison labels at the beginning (determined by `initial_comparison_frac`) and then gather the same number of labels in each iteration. In contrast, the [DRLHP paper](https://arxiv.org/pdf/1706.03741.pdf) uses a schedule where the rate of collecting new labels decreases over time, e.g. for Atari:

> We collect 500 comparisons from a randomly initialized policy network at the beginning of training, and our rate of labeling after T frames of training is decreased every 5 ∗ 10^6 frames, to be roughly proportional to 5 ∗ 10^6/(T + 5 ∗ 10^6).

It makes sense that this would help (after a while, we've learned a good reward model and just need to fine-tune as the policy changes). I'd guess it's less important than #389 but would still be good to have implemented.

The most flexible choice would be to pass a scheduler function to `PreferenceComparisons`, but it would be nice to make it easy to use either a constant schedule (like we currently have) or a hyperbolic schedule (like in the paper), in both cases combined with a configurable number of additional initial steps. (e.g. we could have a `make_hyperbolic_schedule(decrease_every: int, initial_comparison_frac: float) -> Callable[[int], float]` helper function).

The other design decision is how to specify the total number of comparisons we gather. From an end-user perspective, I think it makes much more sense to continue specifying this as a single number (`total_comparisons`) rather than implicitly via the scheduler. So I think we need the scheduler function to be normalized (i.e. its outputs should sum to one over all iterations). Might be best to just do this inside `PreferenceComparisons` by calling the scheduler for all iterations right at the beginning (since the scheduler itself won't know about the total number of iterations).
